### PR TITLE
feat(cache): add `staleMaxAge` option for caching header

### PR DESCRIPTION
**Pull request by Intevel**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control
`stale-while-revalidate` gives you the option to revalidate the response in the background after a certain time.

Example:
`{ swr: true, name: "mySWRData", magAge: 1, staleMaxAge: 30 }`

Will create this `cache-control` headers:
`cache-control: s-maxage=1, stale-while-revalidate=30`

### 📝 Checklist
- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/runtime/cache.ts [hunk 1]
```diff
@@ -135,7 +136,11 @@ export function defineCachedEventHandler (handler: CompatibilityEventHandler, op
       if (opts.magAge) {
         cacheControl.push(`s-maxage=${opts.magAge}`)
       }
-      cacheControl.push('stale-while-revalidate')
+	  if (opts.staleMaxAge) {
+		  cacheControl.push(`stale-while-revalidate=${opts.staleMaxAge}`)
```
Comment (pi0) [inline src/runtime/cache.ts:140]: I was thinking since we are keeping stale value up to `maxAge` anyway does it make sense to default it to `opts.maxAge`?

Comment (Intevel) [reply src/runtime/cache.ts:140]: Mostly you don't use the same values. I think leaving multiple options open to a developer is better.

Comment (pi0) [reply src/runtime/cache.ts:140]: I see, however, this is the actual behavior of cachify function right now. We only revalidate until maxAge reaches not before.

Comment (Intevel) [reply src/runtime/cache.ts:140]: Yes but that is not the point behind `swr`

Comment (pi0) [reply src/runtime/cache.ts:140]: `stale-while-revalidate` header alone tells browser always revalidate no?

Comment (Intevel) [reply src/runtime/cache.ts:140]: Yes, but with the options you can define when it should revalidate and not only when the response expired.

Comment (pi0) [reply src/runtime/cache.ts:140]: I know but we _already know_ that data is not going to be invalidated until maxAge. But really we can improve default generated value later ;)

Comment (Intevel) [reply src/runtime/cache.ts:140]: You have convinced me.

**Comment by pi0** [comment]
Let's iterate with next PRs :)

**Pull request merged**